src/vmq_ranch.erl

```diff
@@ -284,4 +294,12 @@ port_cmd_({ssl, Socket}, Data) ->
 port_cmd_(Socket, Data) ->
     erlang:port_command(Socket, Data).
 
+system_continue(_, _, {State, _NrOfChildren}) ->
+	loop(State).
+
+-spec system_terminate(any(), _, _, _) -> no_return().
+system_terminate(Reason, _, _, {State, _NrOfChildren}) ->
```

Comment:
system_terminate is required.. but the tuple {State, _NrOfChildren} in system_continue and system_terminate is specific to the vmq_queue_sup. You should use just State there.

See: http://erlang.org/doc/design_principles/spec_proc.html